pkg/job/maxdimassociator/associator.go

```diff
@@ -214,6 +216,15 @@ func buildLabelsMap(cwMetric *model.Metric, regexpMapping *dimensionsRegexpMappi
 				}
 			}
 
+			// AWS Sagemaker endpoint name may have upper case characters
+			// Resource ARN is only in lower case, hence transforming
+			// endpoint name value to be able to match the resource ARN
+			if cwMetric.Namespace == "AWS/SageMaker" && name == "EndpointName" {
+				if awsSageMakerEndpointUpper.MatchString(value) {
+					value = strings.ToLower(value)
```

Comment:
Wondering if we can always just convert to lowercase without the regex match?

Reply:
To be honest I added it last minute to match the AmazonMQ one. I thought it would be more efficient to just convert to lowercase when needed but after some tests it is more efficient to just run the conversion every time.
I'll remove that check asap.

Reply:
Regex has been removed now